scipy/sparse/linalg/dsolve/linsolve.py

```diff
@@ -383,3 +384,108 @@ def solve(b):
         return solve
     else:
         return splu(A).solve
+
+
+def spsolve_triangular(A, b, lower=True, copy_A=True, overwrite_b=False):
+    """
+    Solve the equation `A x = b` for `x`, assuming A is a triangular matrix.
+
+    Parameters
+    ----------
+    A : (M, M) sparse matrix
+        A sparse square triangular matrix. Should be in CSR format.
+    b : (M,) or (M, N) array_like
+        Right-hand side matrix in `A x = b`
+    lower : bool, optional
+        Whether `A` is a lower or upper triangular matrix.
+        Default is lower triangular matrix.
+    copy_A : bool, optional
```

Comment:
It seems a little strange to have `copy_A` and `overwrite_b`. Looking at the [scipy.linalg docs](https://docs.scipy.org/doc/scipy-0.18.1/reference/linalg.html), it seems pretty standard to have `overwrite_a=False, overwrite_b=False`. I know that we're not _really_ overwriting `A`, though, so I'm conflicted.